src/ucp/core/ucp_context.c

```diff
@@ -362,11 +387,13 @@ static ucs_status_t ucp_add_tl_resources(ucp_context_h context, ucp_tl_md_t *md,
     /* copy only the resources enabled by user configuration */
     context->tl_rscs = tmp;
     for (i = 0; i < num_tl_resources; ++i) {
-        if (ucp_is_resource_enabled(&tl_resources[i], config, masks)) {
-            context->tl_rscs[context->num_tls].tl_rsc   = tl_resources[i];
-            context->tl_rscs[context->num_tls].md_index = md_index;
+        if (ucp_is_resource_enabled(&tl_resources[i], config, masks, &is_aux)) {
```

Comment:
maybe ucp_is_resource_enabled() could return flags instead of is_aux?
would make code shorter and easier to add more suffixes in the future..